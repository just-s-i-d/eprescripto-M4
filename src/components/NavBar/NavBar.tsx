import React, { SetStateAction, useState } from "react";
import { Link } from "react-router-dom";

import { Avatar, Badge, Popover, Button, Space } from "antd";
import {
  UserOutlined,
  MenuUnfoldOutlined,
  MenuFoldOutlined,
} from "@ant-design/icons";
import ProfilePopOverContent from "../ui/ProfilePopOverContent";
import { Header } from "antd/es/layout/layout";

interface navBarProps {
  collapsed: boolean;
  setCollapsed: React.Dispatch<SetStateAction<boolean>>;
  colorPrimaryBg: string;
  colorText: string;
}
const NavBar = ({
  collapsed,
  setCollapsed,
  colorPrimaryBg,
  colorText,
}: navBarProps) => {
  const [open, setOpen] = useState(false);

  return (
    <Header
      className={`px-6 sticky top-0 z-10 flex items-center justify-between h-[8vh]`}
      style={{ backgroundColor: colorPrimaryBg, color: colorText }}
    >
      <span className="text-3xl text-white">
        <Button
          type="text"
          icon={collapsed ? <MenuUnfoldOutlined /> : <MenuFoldOutlined />}
          onClick={() => setCollapsed(!collapsed)}
          className="text-sm text-white w-8 h-8"
        />
        <Link to="/">ePrescripto</Link>
      </span>
      <Space>
        <Popover
          content={<ProfilePopOverContent />}
          trigger="click"
          open={open}
          onOpenChange={(value: boolean) => setOpen(value)}
        >
          <Badge count={0} size="small">
            <Avatar size={42} icon={<UserOutlined />} />
          </Badge>
        </Popover>
      </Space>
    </Header>
  );
};
export default NavBar;
