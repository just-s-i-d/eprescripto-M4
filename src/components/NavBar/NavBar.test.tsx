import { fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import NavBar from "./NavBar";
import "@test/matchMedia";
import { MemoryRouter } from "react-router-dom";

describe("NavBar component tests", () => {
  const mockedFunction = jest.fn();
  it("Test to check if the Navbar renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <NavBar collapsed={false} setCollapsed={mockedFunction} />
      </MemoryRouter>,
    );
    expect(screen.getByText("ePrescripto")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("NavBar side menu button test", () => {
    render(
      <MemoryRouter>
        <NavBar collapsed={false} setCollapsed={mockedFunction} />
      </MemoryRouter>,
    );
    const sideMenuBtn = screen.getByTestId("side-menu-btn");
    fireEvent.click(sideMenuBtn);
    expect(mockedFunction).toHaveBeenCalled();
  });
});
