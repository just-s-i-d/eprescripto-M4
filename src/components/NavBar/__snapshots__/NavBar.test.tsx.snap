// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component tests Test to check if the Navbar renders correctly 1`] = `
<DocumentFragment>
  <header
    class="ant-layout-header px-6 sticky top-0 z-10 flex items-center justify-between h-[8vh] css-dev-only-do-not-override-1adbn6x"
  >
    <span
      class="text-3xl text-white"
    >
      <button
        class="ant-btn css-dev-only-do-not-override-1adbn6x ant-btn-text ant-btn-icon-only text-sm text-white w-8 h-8"
        data-testid="side-menu-btn"
        type="button"
      >
        <span
          class="ant-btn-icon"
        >
          <span
            aria-label="menu-fold"
            class="anticon anticon-menu-fold"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="menu-fold"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M408 442h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8zm-8 204c0 4.4 3.6 8 8 8h480c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8H408c-4.4 0-8 3.6-8 8v56zm504-486H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zm0 632H120c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-56c0-4.4-3.6-8-8-8zM115.4 518.9L271.7 642c5.8 4.6 14.4.5 14.4-6.9V388.9c0-7.4-8.5-11.5-14.4-6.9L115.4 505.1a8.74 8.74 0 000 13.8z"
              />
            </svg>
          </span>
        </span>
      </button>
      <a
        href="/"
      >
        ePrescripto
      </a>
    </span>
    <div
      class="ant-space css-dev-only-do-not-override-1adbn6x ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
    >
      <div
        class="ant-space-item"
      >
        <span
          class="ant-badge css-dev-only-do-not-override-1adbn6x"
        >
          <span
            class="ant-avatar ant-avatar-circle ant-avatar-icon css-dev-only-do-not-override-1adbn6x"
            style="width: 42px; height: 42px; line-height: 42px; font-size: 21px;"
          >
            <span
              aria-label="user"
              class="anticon anticon-user"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="user"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                />
              </svg>
            </span>
          </span>
        </span>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
